src/pages/product/[id].tsx

```diff
@@ -34,16 +35,30 @@ export default function ProductDetail({
   error: { message: string };
 }) {
   const [totalPrice, setTotalPrice] = useState<number>(product.price || 0);
+  const [quantity, setQuantity] = useState<number>(1);
 
-  const handlePriceChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const newTotalPrice = calculateTotalPrice(e, product.price);
+  const handleAddToCart = () => {
+    addToCart(product, quantity);
+    alert('상품이 장바구니에 추가되었습니다.');
+  };
+
+  const handleQuantityChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setQuantity(Number(e.target.value));
```

Comment:
NIT: 저는 이벤트를 처리할때는 별도로 변수를 만드는게 좋더라고요
```suggestion
    const newQuantitiy = Number(e.target.value);
    setQuantity(newQuantitiy);
```